src/prefect/core/task.py

```diff
@@ -415,6 +415,19 @@ def copy(self, **task_args: Any) -> "Task":
 
         return new
 
+    @property
+    def __signature__(self) -> inspect.Signature:
+        """Dynamically generate the signature, replacing ``*args``/``**kwargs``
+        with parameters from ``run``"""
+        if not hasattr(self, "_cached_signature"):
+            sig = inspect.Signature.from_callable(self.run)
+            parameters = list(sig.parameters.values())
+            parameters.extend(EXTRA_CALL_PARAMETERS)
+            self._cached_signature = inspect.Signature(
+                parameters=parameters, return_annotation="Task"
```

Comment:
The return annotation confused me for a second - we _might_ want to include a small comment in the docstring for `__signature__` that this is the signature of `__call__` not `run`, although now that I write it out it should be obvious -- you can choose to ignore this comment if you want

Reply:
Yeah, since the `__signature__` is attached to the object, it's the signature for calling that object, so it has to be for `__call__` not `run`.